include/irr/asset/IMeshPacker.h

```diff
@@ -0,0 +1,123 @@
+        
+        m_alctrReservedSpace[1] = static_cast<uint8_t*>(_IRR_ALIGNED_MALLOC(core::GeneralpurposeAddressAllocator<uint32_t>::reserved_size(alignof(uint16_t), allocParams.indexBuffSupportedSizeInBytes / sizeof(uint16_t), allocParams.indexBufferMinAllocSize), _IRR_SIMD_ALIGNMENT));
+        m_idxBuffAlctr = core::GeneralpurposeAddressAllocator<uint32_t>(m_alctrReservedSpace[1], 0u, 0u, alignof(uint16_t), allocParams.indexBuffSupportedSizeInBytes / sizeof(uint16_t), allocParams.indexBufferMinAllocSize);
+
+        //m_alctrReservedSpace[2] = _IRR_ALIGNED_MALLOC(core::GeneralpurposeAddressAllocator::reserved_size(/**/), _IRR_SIMD_ALIGNMENT);
+        //m_perInsVtxBuffAlctr = core::GeneralpurposeAddressAllocator(m_alctrReservedSpace[2], 0u, 0u, 0u, allocParams.perInstanceVertexBuffSupportedSizeInBytes / allocParams.perInstanceVertexbufferMinAllocSize, allocParams.perInstanceVertexbufferMinAllocSize);
+
+        //init MDI data allocator
+        m_alctrReservedSpace[3] = static_cast<uint8_t*>(_IRR_ALIGNED_MALLOC(core::GeneralpurposeAddressAllocator<uint32_t>::reserved_size(alignof(uint32_t), allocParams.MDIDataBuffSupportedSizeInBytes / sizeof(MDIStructType), allocParams.MDIDataBuffMinAllocSize), _IRR_SIMD_ALIGNMENT));
+        m_MDIDataAlctr = core::GeneralpurposeAddressAllocator<uint32_t>(m_alctrReservedSpace[3], 0u, 0u, alignof(uint32_t), allocParams.MDIDataBuffSupportedSizeInBytes / sizeof(MDIStructType), allocParams.MDIDataBuffMinAllocSize);
+
+        //1 attrib enabled == 1 binding
+        for (uint16_t attrBit = 0x0001, location = 0; location < 16; attrBit <<= 1, location++)
+        {
+            if (m_outVtxInputParams.enabledAttribFlags & attrBit)
+            {
+                m_outVtxInputParams.bindings[location].stride = getTexelOrBlockBytesize(static_cast<E_FORMAT>(m_outVtxInputParams.attributes[location].format));
+                m_outVtxInputParams.bindings[location].inputRate = preDefinedLayout.bindings[preDefinedLayout.attributes[location].binding].inputRate;
+            }
+        }
+    }
+
+    virtual PackedMeshBufferData alloc(const MeshBufferType const* meshBuffer) = 0;
+    virtual void commit(/*TODO*/) = 0;
+
+protected:
+    virtual ~IMeshPacker() 
+    {
+        for (int i = 0; i < 4; i++)
+            _IRR_ALIGNED_FREE(m_alctrReservedSpace[i]);
+    }
+
+protected:
+    //output mesh buffers data
+    SVertexInputParams m_outVtxInputParams;
+
+    uint8_t* m_alctrReservedSpace[4];
+    core::GeneralpurposeAddressAllocator<uint32_t> m_vtxBuffAlctr;
+    core::GeneralpurposeAddressAllocator<uint32_t> m_idxBuffAlctr;
+    core::GeneralpurposeAddressAllocator<uint32_t> m_perInsVtxBuffAlctr;
+    core::GeneralpurposeAddressAllocator<uint32_t> m_MDIDataAlctr;
+
+    const uint16_t m_maxIndexCountPerMDIData;
```

Comment:
you need a min and max

Reply:
why min?

Reply:
because eventually when you will do clustering of triangles you will need to split before you reach the max, but never split before you reach the minimum (GPUs SIMD lanes are 32 or 64 invocations, you want to ensure they are full or close to full)